### PR TITLE
Components: Remove textarea field global styles

### PR DESCRIPTION
**Pull request by tyxla**
This PR removes global form `textarea` styles in favor of more specific component-level styling. It involves a lot of changes, simply because textarea fields are very common throughout Calypso. This means this PR also requires some thorough testing, so I'd really appreciate some help with that 🙇‍♂️. 

#### Changes proposed in this Pull Request

* Remove global form `textarea` styles in favor of more specific component-level styling. See #45259

#### Testing instructions

(bear with me, this is a long list 😅 )

Test textarea fields at the following locations against production and ensure that they match stylistically and functionally:

* `/comments/all/:site` - for a site with comments:
  * Click "Reply" to open the comment form for replying to a comment
  * Click "Edit" to open the comment editing form
* Reader - open a post with comments and:
  * Observe the field for posting a new comment.
  * Observe the field for replying to an existing comment.
* Reader - edit a list and observe its description field in `/read/list/:username/:listSlug/edit` - cc @Automattic/reader 
* `/devdocs/blocks/user-mentions` - the textarea field
* `/devdocs/design/form-fields` - the textarea fields at the bottom
* `/devdocs/design/counted-textareas` - the textarea field
* `/devdocs/design/textarea-autosize` - the textarea field
* Site Settings:
  * `/settings/discussion/:site` - there are several fields at the bottom
  * `/settings/jetpack/:site` on a Jetpack site, the Private Key textarea field
  * `/settings/:site` on Jetpack Cloud, the Private Key textarea field. I could use someone from @Automattic/jetpack-voyager to verify this works well.
  * `/settings/security/:site` on a Jetpack site, the "Allowed IP addresses" field in the "Prevent brute force login attacks" section when you expand the accordion.
  * `/settings/podcasting/:site` on a paid plan, the podcasting summary field.
* Happychat on any location that has Help enabled - enable yourself as a chat agent in https://hud-staging.happychat.io/ and initiate a chat, then compare the textarea for writing a chat message. Anyone from @Automattic/lighthouse to give this a test?
* `/settings/taxonomies/category/:site` - editing a category, the Description field.
* `/domains/manage/:site/dns/:site` - adding a TXT DNS record, the "Text" textarea field.
* `/me/purchases/:domain/:purchaseId/confirm-cancel-domain` where `:domain` is a domain you have under your WP.com account, and `:purchaseId` is the ID of the purchase of that domain. Select "Something not listed here" and observe the reason textarea field below. Anyone from @Automattic/cobalt to test this and the above domain features?
* Canceling a paid plan purchase in `/me/purchases` - in the last step, the "What's one thing we could have done better? (optional)" textarea field.
* `/me` - the About Me field
* `/help/contact` - the "How can we help?" field. Test that also by going to `/me`, clicking on the question mark on the bottom right, clicking the "Contact Us" button at the bottom, and observing that same textarea field.
* `/me/concierge/:site/book` for a site where you have available Concierge sessions or have just purchased a Business plan. Follow through the steps and observe the "What are you hoping to accomplish with your site?" field.
* Community translator - follow the instructions in #46088, cc @yuliyan and @Automattic/i18n to test this
* NPS Survey: go to `/posts/:site`, type `dispatch( { type: 'NPS_SURVEY_SET_ELIGIBILITY', isSessionPicked: true } )` in your browser console, and the survey will appear. Select some number, go to the next step, and observe the reason textarea field.
* Classic Calypso Editor - if you need to enable it, `return next()` at the beginning of the `gutenberg()` method in the `post-editor/controller`, then remove `/block-editor` from the URL of a post you're editing. Test the following:
  * Adding a post excerpt in the More Options accordion in the sidebar.
  * Editing the Publicize message in the Sharing accordion in the sidebar.
  * Editing the description and caption of a media item in the media modal.
  * Editing the main post content
  * Editing a simple payments form - the description field
  * Previewing a contact form - the "Comment" textarea field
* WooCommerce for a store that has the WooCommerce and WooCommerce Services plugins installed (cc @c-shultz and @timmyc for testing this) - a bunch of areas includes textarea fields:
  * Orders
  * Products
  * Reviews
  * Settings
 * WP Super Cache - install and enable the plugin on a Jetpack site, and go to `/extensions/wp-super-cache/advanced/:site`; there are multiple textarea fields on that page.
 * Zone Manager (Zoninator) - install and enable the plugin on a Jetpack site, and go to `/extensions/zoninator/:site` then create or edit a zone, observing the zone description field.

### CODE REVIEW

**Review by yuliyan — COMMENTED** [review]
Tested community translator component and confirmed it looks fine.

File: client/components/forms/form-textarea/style.scss
Comment (tyxla) [reply client/components/forms/form-textarea/style.scss]: Nice catch! Fixed in https://github.com/Automattic/wp-calypso/pull/46196/commits/117be8876c4cf3379fe7d853c2d481a7006c164d

**Comment by timmyc** [comment]
@tyxla thanks for the ping, I'll give this a spin on the WooCommerce/Store feature.

**Comment by unDemian** [comment]
> Happychat on any location that has Help enabled - enable yourself as a chat agent in https://hud-staging.happychat.io/ and initiate a chat, then compare the textarea for writing a chat message. Anyone from @Automattic/lighthouse to give this a test?

There is a border shadow on focus that was not there before https://d.pr/v/dQeiFB, cc: @evilluendas

**Comment by tyxla** [comment]
> > Happychat on any location that has Help enabled - enable yourself as a chat agent in https://hud-staging.happychat.io/ and initiate a chat, then compare the textarea for writing a chat message. Anyone from @Automattic/lighthouse to give this a test?
> 
> There is a border shadow on focus that was not there before https://d.pr/v/dQeiFB, cc: @evilluendas

Great catch @unDemian! Fixed in https://github.com/Automattic/wp-calypso/pull/46196/commits/28dbad2edf37c5cb853b4c4eacd3de5ed29cf9a1

**Review by sarayourfriend — APPROVED** [review]
I tested anything you hadn't cc'ed a specific team/person to test and it LGTM!

**Comment by tyxla** [comment]
Would be great if we can get a review from the rest of the pinged teams. Should be a quick one, given that there are only a few things to test for each team.

**Review by klimeryk — APPROVED** [review]
Tested from domains side along with @kriskorn  - LGTM!

**Review by mattgawarecki — APPROVED** [review]
Approving for Jetpack Voyager. cloud.jetpack.com only has the one textarea you mentioned, and it looks/behaves exactly as production. Good work! 🚢

**Comment by tyxla** [comment]
Thanks, everyone for the reviews 🙌 

The last one I'd love to get is from @timmyc or @c-shultz for the WooCommerce store extension in Calypso.

**Comment by tyxla** [comment]
I've rebased this after merging #46479 - we're now able to completely remove the `_forms.scss` file 🎉

**Review by timmyc — APPROVED** [review]
Text area in the store section appears to be functioning well on this branch for me.

**Comment by tyxla** [comment]
Thanks, everyone for the reviews 🙌  You're the best!

**Pull request merged**
